nomos-services/data-availability/indexer/src/storage/adapters/rocksdb.rs

```diff
@@ -0,0 +1,175 @@
+        if reply_rx.await?.is_none() {
+            return Ok(());
+        }
+
+        let vid_key = key_bytes(
+            DA_VID_KEY_PREFIX,
+            [app_id.clone().as_ref(), idx.as_ref()].concat(),
+        );
+
+        // We are only persisting the id part of VID, the metadata can be derived from the key.
+        let value = Bytes::from(vid.certificate_id().to_vec());
+
+        self.storage_relay
+            .send(StorageMsg::Store {
+                key: vid_key,
+                value,
+            })
+            .await
+            .map_err(|(e, _)| e.into())
+    }
+
+    async fn get_range_stream(
+        &self,
+        app_id: <Self::VID as Metadata>::AppId,
+        index_range: Range<<Self::VID as Metadata>::Index>,
+    ) -> Box<dyn Stream<Item = (<Self::VID as Metadata>::Index, Option<Bytes>)> + Unpin + Send>
+    {
+        let futures = FuturesUnordered::new();
+
+        // TODO: Using while loop here until `Step` trait is stable.
+        //
+        // For index_range to be used as Range with the stepping capabilities (eg. `for idx in
+        // item_range`), Metadata::Index needs to implement `Step` trait, which is unstable.
+        // See issue #42168 <https://github.com/rust-lang/rust/issues/42168> for more information.
+        let mut current_index = index_range.start.clone();
+        while current_index <= index_range.end {
+            let idx = current_index.clone();
+            let app_id = app_id.clone();
+
+            let (reply_tx, reply_rx) = tokio::sync::oneshot::channel();
+            let key = key_bytes(
+                DA_VID_KEY_PREFIX,
+                [app_id.as_ref(), current_index.as_ref()].concat(),
+            );
+
+            self.storage_relay
+                .send(StorageMsg::Load {
+                    key,
+                    reply_channel: reply_tx,
+                })
+                .await
+                .expect("Failed to send load request to storage relay");
+
+            futures.push(async move {
+                match reply_rx.await {
+                    Ok(Some(id)) => (idx, Some(load_blob(app_id.as_ref(), &id).await)),
+                    Ok(None) => (idx, None),
+                    Err(_) => {
+                        tracing::error!("Failed to receive storage response");
+                        (idx, None)
+                    }
+                }
+            });
+
+            current_index = current_index.next();
+        }
+
+        Box::new(futures)
+    }
+}
+
+fn key_bytes(prefix: &str, id: impl AsRef<[u8]>) -> Bytes {
+    let mut buffer = BytesMut::new();
+
+    buffer.extend_from_slice(prefix.as_bytes());
+    buffer.extend_from_slice(id.as_ref());
+
+    buffer.freeze()
+}
+
+// TODO: Rocksdb has a feature called BlobDB that handles largo blob storing, but further
+// investigation needs to be done to see if rust wrapper supports it.
+async fn load_blob(app_id: &[u8], id: &[u8]) -> Bytes {
```

Comment:
This could probably just return an `Option<Bytes>` or a `Result<Bytes, SomeErr>`, as where is used we are expecting an `Option` as well. The the lack of having the file or whatever is handled by the one that use it. 
Like it is now we can end up with a `Some(EmptyBytes)` which makes no sense. Better a `None` or just `EmptyBytes`

Reply:
`load_blob` returns Some(Bytes) only if read succeeds, any other case returns None now.